### DISCUSSION TITLE
Particle importing

### DISCUSSION THREAD

**Discussion by JMByrne1** [epochpic/epoch#755] [Q&A] [answered]
When considering external injection into plasma, can you import an electron distribution into EPOCH? I.e. particle positions and momenta.

**Comment by Status-Mirror** [comment, marked as answer]
Hey @JMByrne1,

You can either import particles from a file, where you specify the injection time, position, momentum and weight for each one individually, or you can specify the temperature and drift of a plasma outside the simulation window, which can be injected over the simulation. Both examples are present in out [injector demos](https://epochpic.github.io/quickstart/basic_examples/injectors.html).

Hope this helps,
Stuart
